.github/scripts/deploy-registry.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+set -o pipefail
+set -u
+
+VERBOSE="${VERBOSE:-0}"
+if [[ "${VERBOSE}" -ne "0" ]]; then
+    set -x
+fi
+
+# List of required environment variables
+required_vars=(
+    "GCLOUD_API_KEY"
+    "GCLOUD_PROD_DEPLOY_SECRET"
+    "GCLOUD_DEV_DEPLOY_SECRET"
+)
+
+# Check if each required variable is set
+for var in "${required_vars[@]}"; do
+    if [[ -z "${!var:-}" ]]; then
+        echo "Error: Environment variable '$var' is not set."
+        exit 1
+    fi
+done
+
+# Trigger a build for dev 
+# DONT EVER SET ANY FLAGS THAT MIGHT PRINT THE URL, AS IT HAS SECRETS IN IT
+curl -X POST "https://cloudbuild.googleapis.com/v1/projects/coder-registry-1/triggers/http-build-registry-v2-dev:webhook?key=${GCLOUD_API_KEY}&secret=${GCLOUD_DEV_DEPLOY_SECRET}" \
+-H "Content-Type: application/json" \
+-d '{}' \
+--fail
```

Comment:
I'm not entirely convinced about this approach. We could maybe instead trigger a build using the `gcloud` CLI (ref: https://cloud.google.com/sdk/gcloud/reference/builds/triggers/run) in combination with the `setup-gcloud` action (https://github.com/google-github-actions/setup-gcloud)

Reply:
cc @deansheather for some additional eyes.

Reply:
Absolutely would prefer to use gcloud CLI rather than calling googleapis from curl. We also should not be using service account credentials as actions secrets, and should use workload identity instead. I can help you set that up with a new SA for this repo @bcpeinhardt

Reply:
Secrets deleted, will update to use the gcloud cli, and yes some help with a service account would be great haha 😎